### PR TITLE
Issue #1599: make sslProfile configuration updatable

### PR DESCRIPTION
**Pull request by kgiusti**
This does not solve #1572. It is part of a series of changes that will address #1572.

### CODE REVIEW

File: src/tls/tls.c [hunk 1]
```diff
@@ -0,0 +1,780 @@
...
+char *qd_tls2_session_get_alpn_protocol(const qd_tls2_session_t *tls_session)
+{
+    char       *protocol = 0;
+    const char *protocol_name;
+    size_t      protocol_name_length;
+
+    assert(tls_session->pn_raw);
+    if (pn_tls_get_alpn_protocol(tls_session->pn_raw, &protocol_name, &protocol_name_length)) {
+        protocol = (char *) qd_calloc(protocol_name_length + 1, sizeof(char));
+        memmove(protocol, protocol_name, protocol_name_length);
```
[the remark this answers is not in the document]
Comment (kgiusti) [reply src/tls/tls.c:465]: Not true: qd_calloc() will explictly call abort() should the memory allocation fail. The only valid reason for returning NULL is if the sizing parameters are zero. Last time I checked sizeof(char) isn't zero, and protocol_name_length is not negative 1.

[the remark this answers is not in the document]
Comment (kgiusti) [reply src/tls/tls.c:465]: Woops - mmmm... this crow is delish.  Can be null if proton routine does not return 0;

File: include/qpid/dispatch/tls.h [hunk 2]
```diff
@@ -0,0 +1,183 @@
+#ifndef __tls_h__
```
Comment (ganeshmurthy) [inline include/qpid/dispatch/tls.h:1]: Can we please rename `tls.h` to `tls_common.h` ?

File: python/skupper_router/management/skrouter.json [hunk 3]
```diff
@@ -668,50 +668,70 @@
...
-                    "create": true
+                    "create": true,
+                    "update": true
```
Comment (kgiusti) [inline python/skupper_router/management/skrouter.json:716]: This field is unused currently by skupper (legacy stuff). Do not allow update.

File: python/skupper_router/management/skrouter.json [hunk 4]
```diff
@@ -668,50 +668,70 @@
...
                 },
                 "uidNameMappingFile": {
                     "type": "string",
                     "description": "The absolute path to the file containing the unique id to display name mapping",
-                    "create": true
+                    "create": true,
+                    "update": true
```
Comment (kgiusti) [inline python/skupper_router/management/skrouter.json:722]: Also unused, remove update capability

File: include/qpid/dispatch/tls.h [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+} qd_tls_type_t;
+
+typedef enum {
+    QD_TLS_CONFIG_MODE_NONE = 0,  // unset
+    QD_TLS_CONFIG_SERVER_MODE,    // Operate as a TLS server (i.e. listener socket)
+    QD_TLS_CONFIG_CLIENT_MODE,    // Operate as an TLS client (i.e. outgoing connections)
+} qd_tls_config_mode_t;
+
+// sslProfile configuration record
+struct qd_ssl2_profile_t {
```
Comment (ganeshmurthy) [inline include/qpid/dispatch/tls.h:52]: I am guessing `qd_ssl2_profile_t` will be renamed to `qd_ssl_profile_t` once the old `qd_ssl_profile_t` is removed ?

**Review by ted-ross — APPROVED** [review]

File: include/qpid/dispatch/tls.h (hunk 5 above)
Comment (kgiusti) [reply include/qpid/dispatch/tls.h:52]: Yes - I can do that on a follow on patch.

File: include/qpid/dispatch/tls.h (hunk 2 above)
Comment (kgiusti) [reply include/qpid/dispatch/tls.h]: Done.

File: python/skupper_router/management/skrouter.json (hunk 3 above)
Comment (kgiusti) [reply python/skupper_router/management/skrouter.json]: Done.

File: python/skupper_router/management/skrouter.json (hunk 4 above)
Comment (kgiusti) [reply python/skupper_router/management/skrouter.json]: Done.

**Review by ganeshmurthy — APPROVED** [review]

**Pull request merged**